cpp/src/parquet/encoding_test.cc

```diff
@@ -862,5 +862,215 @@ TEST_F(DictEncoding, CheckDecodeIndicesNoNulls) {
   CheckDict(actual_num_values, *builder);
 }
 
+// ----------------------------------------------------------------------
+// BYTE_STREAM_SPLIT encode/decode tests.
+
+namespace {
+template <typename DType>
+void TestEncodeDecodeWithBigInput() {
+  const int nvalues = 10000;
+  using T = typename DType::c_type;
+  std::vector<T> data(nvalues);
+  GenerateData<T>(nvalues, data.data(), NULLPTR);
+  std::unique_ptr<TypedEncoder<DType>> encoder =
+      MakeTypedEncoder<DType>(Encoding::BYTE_STREAM_SPLIT);
+  encoder->Put(data.data(), static_cast<int>(data.size()));
+  std::shared_ptr<Buffer> buffer = encoder->FlushValues();
+
+  std::unique_ptr<TypedDecoder<DType>> decoder =
+      MakeTypedDecoder<DType>(Encoding::BYTE_STREAM_SPLIT);
+  decoder->SetData(static_cast<int>(data.size()), buffer->mutable_data(),
+                   static_cast<int>(buffer->size()));
+  std::vector<T> decoded_data(nvalues);
+  int num_decoded_elements = decoder->Decode(decoded_data.data(), nvalues);
+  ASSERT_EQ(nvalues, num_decoded_elements);
+  for (size_t i = 0U; i < decoded_data.size(); ++i) {
+    ASSERT_EQ(data[i], decoded_data[i]);
+  }
+}
+}  // namespace
+
+// Check that the encoder can handle empty input.
+TEST(ByteStreamSplitEncodeDecode, EncodeZeroLenInput) {
+  std::unique_ptr<TypedEncoder<FloatType>> encoder =
+      MakeTypedEncoder<FloatType>(Encoding::BYTE_STREAM_SPLIT);
+  encoder->Put(NULL, 0);
+  ASSERT_EQ(0U, encoder->EstimatedDataEncodedSize());
+  std::shared_ptr<Buffer> encoded_buffer = encoder->FlushValues();
+  ASSERT_EQ(0, encoded_buffer->size());
+}
```

Comment:
Also question re: helper function / reusable test for all encoders

Reply:
I can do this in a separate patch.